data/transactions/logic/assembler.go

```diff
@@ -1399,25 +1399,29 @@ func typecheck(expected, got StackType) bool {
 	return expected == got
 }
 
-var spaces = [256]uint8{'\t': 1, ' ': 1}
+// semi-colon is quite space-like, so include it
+var spaces = [256]bool{'\t': true, ' ': true, ';': true}
```

Comment:
Maybe rename `spaces` to `separators`? I'm surprised that `\n` isn't on the list and would have thought that `;` would be wherever `\n` is.

Reply:
Since the input is inherently line oriented, newlines can't appear.